src/org/infinity/resource/key/ResourceTreeFolder.java

```diff
@@ -212,6 +212,13 @@ public synchronized boolean retainAll(Collection<?> c) {
       return bRet;
     }
 
+    /**
+     * Fixes build on JDK 21+, where both List and SortedSet have a default reversed() method that clashes
+     */
+    public SortedListSet<T> reversed() {
```

Comment:
I think, `@Override` annotation is needed. Also it is probably better to explicitly call some of the default methods, probably `ArrayList.reversed`

Reply:
I initially had a proper override, but since Near Infinity targets Java 8, the `super()` implementation (and thus the `@Override` annotation) isn't available. I believe this sort of stub is the only way to make it build on both JDK 8 and JDK 21+.

Reply:
Ah, yes, make sense. Probably add this as a comment to the code, so no one other will not add that annotation in the future?

Because we cannot reuse implementation just calling `ArrayList.reverse`, maybe just copy that code?